core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -167,7 +256,275 @@ public void setTransportSender(TransportSender transportSender) {
     }
 
     @Override
+    public void setClientConnector(ClientConnector clientConnector) {
+    }
+
+    @Override
     public String getId() {
         return MSF4J_MSG_PROC_ID;
     }
+
+
+    /*
+    Handle WebSocket handshake
+     */
+    private boolean handleWebSocketHandshake(CarbonMessage carbonMessage, Session session)
+            throws WebSocketEndpointAnnotationException {
+        EndpointsRegistryImpl endpointsRegistry = EndpointsRegistryImpl.getInstance();
+            PatternPathRouter.RoutableDestination<Object>
+                    routableEndpoint = endpointsRegistry.getRoutableEndpoint(carbonMessage);
+        try {
+            //If endpoint cannot be found close the connection
+            if (routableEndpoint == null) {
+                throw new IllegalStateException("Cannot find the URI for the endpoint");
+            }
+
+            Method method = new EndpointDispatcher().getOnOpenMethod(routableEndpoint.getDestination());
+            List<Object> parameterList = new LinkedList<>();
+            Map<String, String> paramValues = routableEndpoint.getGroupNameValues();
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters without" +
+                                                                                 " @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
```

Comment:
IMO shouldn't we add new Object() here to avoid possible null pointer exceptions. Consider using new Object()